TumbleBitSetup.Tests/ProtocolTests.cs

```diff
@@ -399,12 +395,9 @@ public bool _ShortN(BigInteger Exp, int shortKeySize, int longKeySize)
             // A case where "shortKeySize"-bits N is used for proving and "longKeySize"-bits is needed for verifying.
             var keyPair = new RsaKey(Exp, shortKeySize);
 
-            var privKey = keyPair._privKey;
-            var pubKey = new RsaPubKey(keyPair);
-
-            byte[][] signature = PermutationTest.Proving(privKey.P, privKey.Q, privKey.PublicExponent, alpha, ps);
+            var signature = keyPair.CreatePermutationTest(setup);
             // passing "longKeySize" as the keySize.
-            return PermutationTest.Verifying(pubKey, signature, alpha, longKeySize, ps);
+            return keyPair.PublicKey.VerifyPermutationTestProof(signature, setup);
```

Comment:
Ok so what I can do, is to make the `keyLength` optional in the Setup. If not specified, then it will use the Modulus bitlength. Would that work ?